### PR TITLE
Feature/eth uri

### PR DESCRIPTION
**Pull request by justinkook**
- Can validate ethereum Uris

URIs based on EIP - 681 / EIP - 831

### CODE REVIEW

File: packages/crypto-wallet-core/src/validation/eth/index.ts [hunk 1]
```diff
@@ -3,6 +3,25 @@ const utils = require('web3-utils');
 
 export class EthValidation implements IValidation {
   validateAddress(_network: string, address: string): boolean {
+    address = this.sanitizeEthereumUri(address);
```
Comment (micahriggan) [inline packages/crypto-wallet-core/src/validation/eth/index.ts]: I'm somewhat worried about this change. We rely on this method to tell us whether an address is valid, so we can pass it into methods that rely on an address being provided. 

If we do this, we lose that and instead it's more of a "contains" a valid address

File: packages/crypto-wallet-core/src/validation/bch/index.ts [hunk 2]
```diff
@@ -3,21 +3,25 @@ const BitcoreCash = require('bitcore-lib-cash');
...
+
+  validateUri(network: string, address: string): boolean {
     // Check if the input is a valid uri or address
     const URICash = BitcoreCash.URI;
     const AddressCash = BitcoreCash.Address;
 
     // Bip21 uri
     let uri, uriAddress;
-    if (/^bitcoincash:|^bchtest:i/.test(address)) {
+    if (/^bitcoincash:|^bchtest:/i.test(address)) {
```
Comment (matiu) [inline packages/crypto-wallet-core/src/validation/bch/index.ts]: Can we use bitcore-lib-cash for this?

Same thing as here:
https://github.com/bitpay/bitcore/pull/2421/files#diff-57413206f34f99cbc83e311f3b62cde2R11

but with bitcore-lib-cash.

File: packages/crypto-wallet-core/test/validation.ts [hunk 3]
```diff
@@ -17,9 +17,10 @@ describe('Address Validation', () => {
 
   // Uri
   const btcUri = 'bitcoin:1NuKwkDtCymgA1FNLUBaUWLD8s4kdKWvgn';
-  const bchUri = 'bitcoincash:qr8uujscckc56ancdkmqnyyl2rx6pnp24gmdfrf8qd';
+  const bchUri = 'bitcoincash:pp8skudq3x5hzw8ew7vzsw8tn4k8wxsqsv0lt0mf3g';
+  const ethUri = 'ethereum:0x37d7B3bBD88EFdE6a93cF74D2F5b0385D3E3B08A?value=123?gasPrice=123?gas=123?gasLimit=123'
```
Comment (matiu) [inline packages/crypto-wallet-core/test/validation.ts]: should the URI be:

ethereum:0x37d7B3bBD88EFdE6a93cF74D2F5b0385D3E3B08A?value=123&gasPrice=123&gas=123&gasLimit=123

could you also add a naked one, like:
ethereum:0x37d7B3bBD88EFdE6a93cF74D2F5b0385D3E3B08A

Comment (justinkook) [reply packages/crypto-wallet-core/test/validation.ts]: added

**Pull request merged**